### DISCUSSION TITLE
[doris-2.0.2] 有关Java UDAF的一些疑问

### DISCUSSION THREAD

**Discussion by JoshuaNocturne** [apache/doris#29255] [A - General / Q&A] [answered]
业务需求上需要实现一个自动聚合json的功能，联想到doris的Aggregate模型和Java UDAF，编写了Java UDF也测试过了，但是作为Aggregate表一直没有测试通过，想问下我们思路是否有问题？doris当前是否支持Aggregate表自动调用java udaf聚合数据？是写时聚合还是读时聚合？

**Comment by LemonLiTree** [comment, marked as answer]
agg算子不支持udaf

**Reply by JoshuaNocturne** [reply]
那只能在查询时调用UDAF？这种场景是否有更好的解决方案呢

**Reply by LemonLiTree** [reply]
对 查询时调用UDAF, 如果性能不行可以 做定期调度，比如每五分钟跑一次，结果写入目标表
